x-pack/legacy/plugins/apm/public/components/shared/Links/apm/ExternalLinks.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { IBasePath } from 'kibana/server';
+import url from 'url';
+
+export const getTraceUrl = (
+  kibanaBasePath: IBasePath,
+  traceId: number | string
+) =>
+  url.format({
+    pathname: kibanaBasePath.prepend('/apm/apm'),
+    hash: `/link-to/trace/${traceId}`
+  });
```

Comment:
@sqren I ended up adding the kibana base path here as parameter. wdyt?

Reply:
I think there are two approaches:

```ts
const traceId = 1337;
const url = getTraceUrl(core.http.basePath, traceId)
```

vs

```ts
const traceId = 1337;
const url = core.http.basePath.prepend(getTraceUrl(traceId));
```

@jasonrhodes do you have any prefs here?

Reply:
@cauemarcondes Let's go for option 2 if you don't mind. The consumer shouldn't have to pass in `basePath` as an argument, but will have to remember to wrap the return value themselves. It makes for a simpler signature.